timestep.c

```diff
@@ -19,6 +19,18 @@
  */
 
 
+/* variables for organizing PM steps of discrete timeline */
+static struct time_vars {
```

Comment:
I suggest changing the name to reflect that.. a struct timespan shall have start and duration, or start and end (exclusive). I think the name `step`  and `PM_Ti` are a bit confusing.